src/allocator/mod.rs

```diff
@@ -0,0 +1,48 @@
+// use crate::arch;
+use crate::*;
+
+// TODO: Find real value
+const PAGE_NUMBER: usize = 50;
+
+static mut PageUsage: [UsageFlags; PAGE_NUMBER] = [UsageFlags::Free; PAGE_NUMBER];
+
+#[repr(u8)]
+#[derive(Debug, PartialEq)]
+enum UsageFlags {
+    Used = 1,
+    Free = 0,
+}
+
+pub fn page_alloc() -> Option<usize> {
+    let mut index = 0;
+
+    // FIXME: Iterator
+    while index < 50 {
```

Comment:
Why 50?

Reply:
I had problems with .bss and .data section. I've removed most of the statics and const to test if it will work. This will be fixed before going out of draft state

Reply:
Oh sorry I hadn't even noticed it was a draft :D My bad!